File: backend/src/main/java/com/demo/aiknowledge/entity/Conversation.java
package com.demo.aiknowledge.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;
import java.time.LocalDateTime;

@Data
@TableName("conversation")
public class Conversation {
    @TableId(type = IdType.AUTO)
    private Long id;
    private Long userId;
    private String title;
    private String scene;
    private Boolean isPinned;
    private LocalDateTime createTime;
    private LocalDateTime updateTime;

    @TableField(exist = false)
    private Integer messageCount;
}

File: android/app/src/main/java/com/evanyao/shopagent/data/model/Conversation.kt
package com.evanyao.shopagent.data.model

data class Conversation(
    val id: Long,
    val userId: Long,
    val title: String? = null,
    val isPinned: Boolean = false,
    val createTime: String? = null,
    val updateTime: String? = null
)
